runtime/compiler/runtime/ClassUnloadAssumption.cpp

```diff
@@ -1280,6 +1292,23 @@ J9::PersistentInfo::addUnloadedClass(
    }
 
 
+void TR_AddressSet::destroy()
+   {
+   jitPersistentFree(_addressRanges);
+   }
+
+void TR_AddressSet::getRanges(std::vector<TR_AddressRange> &ranges)
+   {
+   ranges.insert(ranges.begin(), _addressRanges, _addressRanges + _numAddressRanges);
+   }
+
+void TR_AddressSet::setRanges(std::vector<TR_AddressRange> &ranges)
+   {
+   TR_ASSERT(ranges.size() <= _maxAddressRanges, "Setting too many ranges");
```

Comment:
This is very tricky. `_maxAddressRanges` could be increased at the client with a command line option, so the ASSERT is not enough.
I think that the dynamic allocation of _unloadedClassAddresses would have helped here: have a null pointer and allocate the object only `_maxAddressRanges` is known which happens when the message `getUnloadedClassRanges` is sent in `processUnloadedClasses`.